backend/local/fadvise_unix.go

```diff
@@ -0,0 +1,81 @@
+//+build linux
+
+package local
+
+import (
+	"io"
+	"os"
+
+	"github.com/rclone/rclone/fs"
+	"golang.org/x/sys/unix"
+)
+
+type fadvise struct {
+	o          *Object
+	fd         int
+	lastPos    int64
+	curPos     int64
+	windowSize int64
+}
+
+func (f *fadvise) sequential(limit int64) {
+	l := int64(0)
+	if limit > 0 {
+		l = limit
+	}
+	if err := unix.Fadvise(f.fd, f.curPos, l, unix.FADV_SEQUENTIAL); err != nil {
+		fs.Infof(f.o, "fadvise sequential failed on file descriptor %d: %s", f.fd, err)
+	}
+}
+
+func (f *fadvise) next(n int) {
+	f.curPos += int64(n)
+	f.freePagesIfNeeded()
+}
+
+func (f *fadvise) freePagesIfNeeded() {
+	if f.curPos >= f.lastPos+f.windowSize {
+		f.freePages()
+	}
+}
+
+func (f *fadvise) freePages() {
+	if err := unix.Fadvise(f.fd, f.lastPos, f.curPos-f.lastPos, unix.FADV_DONTNEED); err != nil {
+		fs.Infof(f.o, "fadvise dontneed failed on file descriptor %d: %s", f.fd, err)
+	}
+	f.lastPos = f.curPos
+}
+
+type fadviseReadCloser struct {
+	*fadvise
+	inner io.ReadCloser
+}
+
+const readWindow = 128 * 1024
+
+func newFadviseReadCloser(o *Object, f *os.File, offset, limit int64) io.ReadCloser {
+	r := fadviseReadCloser{
+		fadvise: &fadvise{
+			o:          o,
+			fd:         int(f.Fd()),
+			lastPos:    offset,
+			curPos:     offset,
+			windowSize: readWindow,
+		},
+		inner: f,
+	}
+	r.sequential(limit)
```

Comment:
Should this maybe be bombing out here and just return `f` if `r.sequential` fails?

Reply:
Yes, this is a good idea.